realm/realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -254,6 +259,9 @@ JNIEXPORT void JNICALL Java_io_realm_internal_Table_nativeConvertColumnToNullabl
                 case type_Table:
                     // checked previously
                     break;
+                case type_OldDateTime:
+                    // not used
```

Comment:
Then throwing an exception should be safe :) IMO I would much rather we threw exceptions in these cases as it will catch anyone accidentally using them in the future.